torch/_inductor/scheduler.py

```diff
@@ -246,9 +250,15 @@ def has_aliasing_or_mutation(self) -> bool:
         return bool(self.get_aliases() or self.get_mutations())
 
     def set_read_writes(self, rw: dependencies.ReadWrites) -> None:
-        self.read_writes: dependencies.ReadWrites = rw
+        self.read_writes = rw
         self.unmet_dependencies = self.read_writes.reads
         self.prune_deps()
+        # common_deps are a summary of read_writes used by score_fusion_memory()
+        self.common_deps = {
+            (dep, hint)
```

Comment:
NIT: This would make more sense as a dictionary `dep_to_numbytes_hint`. As lookup by hint doesn't make sense and at this stage there is nothing "common" as it only refers to a single node.

Reply:
"common" refers to the fact that it's both read & write (maybe `all_deps` or `score_fusion_memory_deps` would be a better name?)

Using a dictionary would be slower because we need to find the intersection between the deps of two nodes - so if it were a dict we'd have to construct the `keys()` set at the point of comparison.

Reply:
> "common" refers to the fact that it's both read & write (maybe all_deps or score_fusion_memory_deps would be a better name?)

No, common refers to it being the dependencies that the two nodes have in common with each other. The new name is fine though.

> Using a dictionary would be slower because we need to find the intersection between the deps of two nodes - so if it were a dict we'd have to construct the keys() set at the point of comparison.

Hrm okay, I didn't know that dict keys objects didn't support union natively.

Reply:
I imagine that they do - but you then have to construct keys objects in O(n^2) calls to `score_fusion_memory_deps`.